src/main.rs

```diff
@@ -463,25 +483,64 @@ fn run_cli(cli: Cli) -> Result<()> {
                 };
                 row.commit()?;
             }
-            RecordsTableRow::Missing(missing) => {
-                missing.commit()?;
-                error!("Citation key '{citation_key}' does not exist.");
+            State::NullRemote(remote_id, null_row) => match report {
+                InfoReportType::All => {
+                    println!("Null record: {remote_id}");
+                    let null_row_data = null_row.apply(row::get_null_row_data)?;
+                    println!("Last attempted: {}", null_row_data.attempted);
+                }
+                InfoReportType::Canonical => {
+                    bail!("No canonical id for null record '{remote_id}'");
+                }
+                InfoReportType::Valid => {
+                    bail!("Null record '{remote_id}' is automatically invalid");
+                }
+                InfoReportType::Equivalent => {
+                    bail!("No equivalent keys for null record '{remote_id}'");
+                }
+                InfoReportType::Modified => {
+                    println!("{}", null_row.apply(row::get_null_attempted)?);
+                }
+            },
+            State::UndefinedRemote(remote_id, undefined) => {
+                undefined.commit()?;
+                bail!("Cannot obtain report for record not in database: '{remote_id}'");
             }
+            State::UndefinedAlias(alias) => match report {
+                InfoReportType::All => {
+                    println!("Undefined alias: {alias}");
+                }
+                InfoReportType::Canonical => {
+                    bail!("No canonical id for undefined alias '{alias}'");
+                }
+                InfoReportType::Valid => {
+                    if !is_entry_key(alias.name()) {
+                        error!("Invalid BibTeX: {}", alias.name());
+                    }
+                }
```

Comment:
Is it reasonable to check validity even when the alias is not actually in the database? I can easily imagine this use-case; just wondering if this is a sensible return type.

Reply:
This might be misleading (by not telling the user that the alias doesn't exist). I think `autobib info <alias> valid` is a bad way to check whether an alias is a valid citation key.

Also, we should make the info report type something like a bit flag.

Reply:
I don't know if it's possible for this to be a bit flag; what does it mean to ask for 'canonical' and 'valid' at the same time? I guess there is the "report" itself, and then the methods to access various parts.